fv/config_update_test.go

```diff
@@ -35,32 +35,35 @@ import (
 var _ = Context("Config update tests, after starting felix", func() {
 
 	var (
-		etcd            *containers.Container
-		felix           *containers.Container
-		felixInitialPID int
-		client          client.Interface
-		w               [3]*workload.Workload
+		etcd     *containers.Container
+		felix    *containers.Container
+		felixPID int
+		client   client.Interface
+		w        [3]*workload.Workload
 	)
 
 	getFelixPIDs := func() []int {
 		return felix.GetPIDs("calico-felix")
 	}
 
-	BeforeEach(func() {
-		felix, etcd, client = containers.StartSingleNodeEtcdTopology()
-
+	updateFelixPID := func() {
 		// Get Felix's PID.  This retry loop ensures that we don't get tripped up if we see multiple
-		// PIDs, which can happen transiently when Felix forks off a subprocess.
+		// PIDs, which can happen transiently when Felix restarts/forks off a subprocess.
 		start := time.Now()
 		for {
 			pids := getFelixPIDs()
-			Expect(pids).NotTo(BeEmpty())
 			if len(pids) == 1 {
-				felixInitialPID = pids[0]
+				felixPID = pids[0]
 				break
 			}
 			Expect(time.Since(start)).To(BeNumerically("<", time.Second))
+			time.Sleep(50 * time.Millisecond)
```

Comment:
Logic here doesn't feel quite right to me.  Shouldn't it be 'wait until there's 1 Felix PID that is stable for 0.5s' ?  As it stands, I think the code can be fooled by seeing a PID of a Felix on the verge of restarting.

Reply:
That's OK, I think: we want to get the old PID, then the tests use that to see whether felix restarted or not.

When we call this update method in the new test, it's after waiting for the old PID to disappear.